core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -577,6 +586,38 @@ private synchronized QueryResultRows removePagesFromExchange(ResultQueryInfo que
         return resultBuilder.build();
     }
 
+    private static long estimateJsonSize(Page page)
```

Comment:
@wendigo I'm not sure if we really need this.  Of the 3 usages, this is the only one where I could see it mattering (assuming logical size was actually implemented correctly).

Reply:
I'll take a look

Reply:
I'll take a look